test/fast/Unit tests/nvm_ls_zsh

```diff
@@ -0,0 +1,46 @@
+# This file tests some zsh specific functionality, in automated tests
+# it is called with the shell directly and the hashbang is not used.
+# For manual running though, call with zsh/bash/sh etc.
```

Comment:
i think it'd be simpler to make a normal `nvm_ls` test, and then only enable the various opts when it's zsh. I don't want any shell-specific test files.

Reply:
Updated